nntrainer/dataset/random_data_producers.cpp

```diff
@@ -0,0 +1,167 @@
+// SPDX-License-Identifier: Apache-2.0
+/**
+ * Copyright (C) 2021 Jihoon Lee <jhoon.it.lee@samsung.com>
+ *
+ * @file   random_data_producers.cpp
+ * @date   09 July 2021
+ * @brief  This file contains various random data producers
+ * @see    https://github.com/nnstreamer/nntrainer
+ * @author Jihoon Lee <jhoon.it.lee@samsung.com>
+ * @bug    No known bugs except for NYI items
+ *
+ */
+#include <random_data_producers.h>
+
+#include <base_properties.h>
+#include <node_exporter.h>
+#include <util_func.h>
+
+namespace nntrainer {
+
+/**
+ * @brief Props containing min value
+ *
+ */
+class PropsMin : public Property<float> {
+public:
+  /**
+   * @brief Construct a new props min object with a default value
+   *
+   * @param value default value
+   */
+  PropsMin(float value = 0.0f) : nntrainer::Property<float>(value) {}
+  static constexpr const char *key = "min"; /**< unique key to access */
+  using prop_tag = float_prop_tag;          /**< property type */
+};
+
+/**
+ * @brief Props containing max value
+ *
+ */
+class PropsMax : public Property<float> {
+public:
+  /**
+   * @brief Construct a new props max object with a default value
+   *
+   * @param value default value
+   */
+  PropsMax(float value = 1.0f) : nntrainer::Property<float>(value) {}
+  static constexpr const char *key = "max"; /**< unique key to access */
+  using prop_tag = float_prop_tag;          /**< property type */
+};
+
+/**
+ * @brief Props containing max value
+ *
+ */
+class PropsDataSize : public Property<unsigned int> {
+public:
+  /**
+   * @brief Construct a new props data size object with a default value
+   *
+   * @param value default value
+   */
+  PropsDataSize(unsigned int value = 512) :
+    nntrainer::Property<unsigned int>(value) {}
+  static constexpr const char *key = "size"; /**< unique key to access */
+  using prop_tag = uint_prop_tag;            /**< property type */
+};
+
+RandomDataOneHotProducer::RandomDataOneHotProducer() :
+  rd_one_hot_props(new Props()) {}
+
+RandomDataOneHotProducer::~RandomDataOneHotProducer() {}
+
+const std::string RandomDataOneHotProducer::getType() const {
+  return RandomDataOneHotProducer::type;
+}
+
+unsigned long long
+RandomDataOneHotProducer::size(const std::vector<TensorDim> &input,
+                               const std::vector<TensorDim> &output) const {
+  return std::get<PropsDataSize>(*rd_one_hot_props).get();
```

Comment:
I wonder the purpose of the property PropsDataSize here? Shouldn't the size be size of input/output?

Reply:
In the implementation here, it seems like it returns "the total number of inputs + outputs pairs which the dataset can provide", but in documentation in #1389, the description seems to give the size of the data the producer will have to allocate/read for each iteration. 
Can you please either update the description or check if this return value is correct.

Reply:
Confirming that size here is number of iteration here, but as per discussion at the last Friday, the behavior of `size()` will likely to change to number of elements it has.

Reply:
Please update the documentation in this PR to match the intended behavior. In the header of data_producers, the description is a bit different or unclear.

Reply:
and also input and output are not relevant here. Is it because it is random?? But even if it is random.. I guess the required size of input is necessary to generate data.

Reply:
Ah, I understood. This has changed to label_dims, also updated the documentation in he data_producers.h

Reply:
For the record, Props data size has been changed to propsNumSamples in the future PR